web2.0/src/Components/Historic/HistData.js

```diff
@@ -0,0 +1,110 @@
+            className={currentData?.windData[i].windSpeed.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].waveHeight.dataLabel}
+            weatherData={currentData?.swellData[i].waveHeight.data}
+            extraData={currentData?.swellData[i].waveHeight.unit}
+            className={currentData?.swellData[i].waveHeight.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].swellHeight.dataLabel}
+            weatherData={currentData?.swellData[i].swellHeight.data}
+            extraData={currentData?.swellData[i].swellHeight.unit}
+            className={currentData?.swellData[i].swellHeight.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].swellPeriod.dataLabel}
+            weatherData={currentData?.swellData[i].swellPeriod.data}
+            extraData={currentData?.swellData[i].swellPeriod.unit}
+            className={currentData?.swellData[i].swellPeriod.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].windWaveHeight.dataLabel}
+            weatherData={currentData?.swellData[i].windWaveHeight.data}
+            extraData={currentData?.swellData[i].windWaveHeight.unit}
+            className={currentData?.swellData[i].windWaveHeight.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].windWavePeriod.dataLabel}
+            weatherData={currentData?.swellData[i].windWavePeriod.data}
+            extraData={currentData?.swellData[i].windWavePeriod.unit}
+            className={currentData?.swellData[i].windWavePeriod.className}
+          />
+          <WeatherCard
+            weatherDataLabel={'Swell Direction'}
+            weatherData={currentData?.swellData[i].swellDirection.data}
+            extraData={currentData?.swellData[i].swellDirection.unit}
+            className={currentData?.swellData[i].swellDirection.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].windWaveDirection.dataLabel}
+            weatherData={currentData?.swellData[i].windWaveDirection.data}
+            extraData={currentData?.swellData[i].windWaveDirection.unit}
+            className={currentData?.swellData[i].windWaveDirection.className}
+          />
+          <WeatherCard
+            weatherDataLabel={currentData?.swellData[i].steepness.dataLabel}
+            weatherData={
+              currentData?.swellData[i].steepness.data === 'N/A'
+                ? 'FLAT'
+                : currentData?.swellData[i].steepness.data
+            }
+            extraData={currentData?.swellData[i].steepness.unit}
+            className={currentData?.swellData[i].steepness.className}
+          />
+        </div>,
+      );
+    }
+    return histData;
+    // end makeHistData
```

Comment:
I like this clever use of comments here. It helps the reader understand the ending of particular function calls.

Consider breaking your code into smaller files when you find yourself compelled to track the end of your functions in that way.

Consider whether it's possible for you to further refactor your data structure to make it even more generic to support that (e.g., columnName as an additional field, so that you can free yourself from the burden of assigning custom keys to your the WeatherCard that you refactored into a single component (that was a well executed refactor btw, I like it <3).